### PR TITLE
Make new types that shouldn't be exposed internal

### PR DESCRIPTION
**Pull request by halter73**
#2544

### CODE REVIEW

File: src/Kestrel.Core/Internal/HttpProtocols.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 using System;
 
-namespace Microsoft.AspNetCore.Server.Kestrel.Core
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
 {
     [Flags]
     public enum HttpProtocols
```
Comment (Tratcher) [inline src/Kestrel.Core/Internal/HttpProtocols.cs]: This one can go full internal until 2.2. No need to change the namespace.

**Review by Tratcher — APPROVED** [review]

File: src/Kestrel.Core/Internal/HttpProtocols.cs (hunk 1 above)
Comment (halter73) [reply src/Kestrel.Core/Internal/HttpProtocols.cs]: The only problem with that is that the UseHttpServer IConnectionBuilder methods will need to also change from pubternal to actually internal. @davidfowl Are you OK with that?

Comment (halter73) [reply src/Kestrel.Core/Internal/HttpProtocols.cs]: For reference, these are all the additional changes that would need to be made to make this internal: https://github.com/aspnet/KestrelHttpServer/commit/27c8ea4e562668ca22ba6c7f69ee47d3df22f10d

Comment (Tratcher) [reply src/Kestrel.Core/Internal/HttpProtocols.cs]: Though most of those namespace changes are reverts.

Comment (halter73) [reply src/Kestrel.Core/Internal/HttpProtocols.cs]: True. Focus on the methods/ctor/and property that are now also internal.

Comment (davidfowl) [reply src/Kestrel.Core/Internal/HttpProtocols.cs]: Ugh, I wanted to demo this thing but thats ok.

Comment (halter73) [reply src/Kestrel.Core/Internal/HttpProtocols.cs]: Or we could just keep it pubternal like it is in this PR.

**Review by Tratcher — APPROVED** [review]

**Review by davidfowl — APPROVED** [review]

**Comment by halter73** [comment]
@dotnet-bot test Ubuntu 16.04 Release Build

**Comment by halter73** [comment]
Not approved for 2.1. We will have to make breaking change announcements for the APIs that will change in 2.2.

**Pull request closed without merging**